src/Api/Endpoints/Candidate/Visit/Qc.php

```diff
@@ -0,0 +1,206 @@
+               );
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return array(
+                'v0.0.2',
+                'v0.0.3-dev',
+               );
+    }
+
+    /**
+     * Handles a request that starts with /candidates/$candid
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $pathparts = $request->getAttribute('pathparts');
+        if (count($pathparts) === 1 && array_pop($pathparts) === 'imaging') {
+            switch ($request->getMethod()) {
+            case 'GET':
+                return $this->_handleGET($request);
+
+            case 'PUT':
+                return $this->_handlePUT($request);
+
+            case 'OPTIONS':
+                return (new \LORIS\Http\Response())
+                    ->withHeader('Allow', $this->allowedMethods());
+
+            default:
+                return new \LORIS\Http\Response\MethodNotAllowed(
+                    $this->allowedMethods()
+                );
+            }
+        }
+        return new \LORIS\Http\Response\NotFound();
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (!isset($this->_cache)) {
+
+            $qcstatus = $this->_visit->getImagingQC();
+
+            $view = (new \LORIS\Api\Views\Visit\Qc($this->_visit, $qcstatus))
+                ->toArray();
+
+            $this->_cache = new \LORIS\Http\Response\JsonResponse($view);
+        }
+        return $this->_cache;
+    }
+
+    /**
+     * Sets new value for this session imaging qc.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request): ResponseInterface
+    {
+        $user = $request->getAttribute('user');
+        $data = json_decode((string) $request->getBody(), true);
+
+        $requiredfields = array(
+                           'Meta',
+                           'SessionQC',
+                           'Pending',
+                          );
+        $diff           = array_diff($requiredfields, array_keys($data));
+        if (!empty($diff)) {
+            return new \LORIS\Http\Response\BadRequest(
+                'Field(s) missing: ' . implode(', ', $diff)
+            );
+        }
+
+        $candid = $data['Meta']['CandID'] ?? null;
```

Comment:
Maybe use CandID object?

Reply:
Yes, but not in this PR. TimePoint::getCandID still returns a integer.

Reply:
This PR: https://github.com/aces/Loris/pull/4350

😄